conda_env/cli/main.py

```diff
@@ -79,6 +79,7 @@ def main():
     initialize_logging()
     parser = create_parser()
     args = parser.parse_args()
+    os.environ["CONDA_AUTO_UPDATE_CONDA"] = "false"
```

Comment:
I think it's reasonable to do this to avoid the most common error cases. So this could be merged as a first step.
But this wouldn't catch all possible cases, e.g.,
 1. `conda =some-version` is actually given in an `environment.yml`,
 2. any of `conda`'s dependencies get updated.

For the second case imagine one of the dependency wasn't completely loaded (for whatever reason) before the Conda packages were updated. After that the subsequent operations could try to load some of the previously unloaded parts => again, you'd have some inconsistent running process..

Reply:
I wonder if #6834 will fix the second issue? 🤔I think it might, given that we try very hard to limit in-line imports.

Reply:
> given that we try very hard to limit in-line imports.

That should make it less likely, yes. But what if a dependency doesn't care that much about in-line imports and thus might hit one in some cases? I'm not sure we can be entirely certain, things will go fine in all cases.